src/PerfView/EventViewer/EventWindow.xaml.cs

```diff
@@ -605,7 +561,143 @@ private void DoShowEventCounterGraph(object sender, ExecutedRoutedEventArgs e)
+                if (name == null)
+                    return false;
+
+                string displayName = GetStringField(rest, DisplayNameToken, ref pos);
+                if (displayName == null)
+                    displayName = name;
+
+                // check for Sum or Mean type of .NET Core counter value
+                var value = GetNumericField(rest, IncrementToken, ref pos);
+                if (value == null)
+                {
+                    value = GetNumericField(rest, MeanToken, ref pos);
+                    if (value == null)
+                        return false;
+                }
+
+                var interval = GetNumericField(rest, IntervalToken, ref pos);
+                if (interval == null)
+                    return false;
+
+                string namePart = displayName;
+                string meanPart = value;
+                string intervalSecPart = interval;
+
+                double mean;
+                double intervalSec;
+                if (!double.TryParse(meanPart, out mean))
+                {
+                    return false;
+                }
+                if (!double.TryParse(intervalSecPart, out intervalSec))
+                {
+                    return false;
+                }
+
+                if (!counters.TryGetValue(namePart, out var points))
+                {
+                    points = new List<Tuple<double, double>>();
+                    counters.Add(namePart, points);
+                }
+                points.Add(Tuple.Create(t, mean));
+
+                t += intervalSec;
+
+                return true;
+            });
+
+            return counters;
+        }
+
+        private string GetStringField(string payload, string token, ref int pos)
+        {
+            var next = pos;
+
+            // a string field is stored in the payload as:
+            //    <token>"<value>"
+            // note that <token> has the following format: <field>=
+            //
+            next = payload.IndexOf(token, next);
+            if (next == -1)
+                return null;
+
+            next += token.Length;
+            if (payload[next] != '"')
+                return null;
+            // skip the " at the beginning of the field value
+            next++;
+
+            var end = payload.IndexOf('"', next);
+            if (end == -1)
+                return null;
+
+            var length = end - next;
+            pos = end;
+            return payload.Substring(next, length);
+
+        }
+
+        private string GetNumericField(string payload, string field, ref int pos)
+        {
+            var next = pos;
+
+            // a numeric field is stored in the payload as:
+            //    <token><value>
+            // note that <token> has the following format: <field>:
+            //
+            next = payload.IndexOf(field, next);
+            if (next == -1)
+                return null;
```

Comment:
Your code works for netcoreapp3.0, but not for desktop.  The reason is that IntervalSec is the last entry in the payload and so there isn't a ',' at the end of it.  With this change, your code works for both netcoreapp3.0 and desktop.

```suggestion
            {
                end = payload.IndexOf(" }", next);
                if (end == -1)
                    return null;
            }
```

Reply:
here is the required change (i.e. not at next level but to compute end)

            var end = payload.IndexOf(',', next);
            // handle the case of the last numeric value of the payload 
            // i.e. look for " }" instead of ","
            if (end == -1)
            {
                end = payload.IndexOf(" }", next);
                if (end == -1)
                    return null;
            }

I've tested it and it works

Reply:
Ah, yes.  You're right.  I put this in the wrong spot.